### PR TITLE
Auth: Add pdns_control command to the the list of XFR domains in queue

### PR DESCRIPTION
**Pull request by pieterlexis**
### Short description
This allows some insight into what the auth is doing

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [x] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)

### CODE REVIEW

File: pdns/common_startup.cc [hunk 1]
```diff
@@ -371,6 +371,7 @@ void declareStats(void)
   S.declare("latency","Average number of microseconds needed to answer a question", getLatency, StatType::gauge);
   S.declare("timedout-packets","Number of packets which weren't answered within timeout set");
   S.declare("security-status", "Security status based on regular polling", StatType::gauge);
+  S.declare("xfr-queue", "Size of the queue of domains to be XFR'd", [](const string&) { return Communicator.getSuckRequestsWaiting(); }, StatType::gauge);
```
Comment (Habbie) [inline pdns/common_startup.cc:374]:
```suggestion
  S.declare("xfr-queue", "Size of the queue of domains to be XFRd", [](const string&) { return Communicator.getSuckRequestsWaiting(); }, StatType::gauge);
```

File: pdns/slavecommunicator.cc [hunk 2]
```diff
@@ -1003,3 +1003,16 @@ void CommunicatorClass::slaveRefresh(PacketHandler *P)
...
+  vector<pair<DNSName, ComboAddress> > ret;
+  std::lock_guard<std::mutex> l(d_lock);
+  for (auto const &d : d_suckdomains) {
+    ret.push_back(make_pair(d.domain, d.master));
+  }
+  return ret;
+}
+
+size_t CommunicatorClass::getSuckRequestsWaiting() {
+  return d_suckdomains.size();
```
Comment (mind04) [inline pdns/slavecommunicator.cc:1017]: No lock here?

File: pdns/slavecommunicator.cc [hunk 3]
```diff
@@ -1003,3 +1003,16 @@ void CommunicatorClass::slaveRefresh(PacketHandler *P)
     }
   }
 }
+
+vector<pair<DNSName, ComboAddress> > CommunicatorClass::getSuckRequests() {
+  vector<pair<DNSName, ComboAddress> > ret;
+  std::lock_guard<std::mutex> l(d_lock);
```
Comment (mind04) [inline pdns/slavecommunicator.cc:1009]:
```suggestion
  std::lock_guard<std::mutex> l(d_lock);
  ret.reserve(d_suckdomains.size());
```

File: pdns/common_startup.cc [hunk 4]
```diff
@@ -371,6 +371,7 @@ void declareStats(void)
   S.declare("latency","Average number of microseconds needed to answer a question", getLatency, StatType::gauge);
   S.declare("timedout-packets","Number of packets which weren't answered within timeout set");
   S.declare("security-status", "Security status based on regular polling", StatType::gauge);
+  S.declare("xfr-queue", "Size of the queue of domains to be XFRd", [](const string&) { return Communicator.getSuckRequestsWaiting(); }, StatType::gauge);
```
Comment (pieterlexis) [inline pdns/common_startup.cc:374]: `[](const string&` => `StatType::gauge` (probably)

Comment (Habbie) [reply pdns/common_startup.cc:374]: ah, it's an unused argument!

**Pull request merged**